add notes for n25380 (#527)
bitcoin-core-review-club/website: _posts/2022-07-13-#25380.md

components: ["mempool"]
host: glozow
status: upcoming
commit: 200a49f
---

## Notes

- Fees for the wallet are calculated using a fee estimator attached to the node's mempool,
  [`CBlockPolicyEstimator`](https://github.com/bitcoin/bitcoin/blob/994573726f81afca5343e2a8d82309e697e61f90/src/policy/fees.h#L132).

    - At a high level, it records the feerate and time at which transactions enter the mempool and

>       how many blocks later they confirm. Later, it uses this information ot provide the opposite

Reviewer: ```suggestion
      how many blocks later they confirm. Later, it uses this information to provide the opposite
```

  into consideration, (see [#23074](https://github.com/bitcoin/bitcoin/pull/23074)), but the
  calculation is not simple.

- [PR #25380](https://github.com/bitcoin/bitcoin/pull/25380) prevents the fee estimator from
  underestimating feerates due to inaccurately recording feerates for CPFP'd transactions.

## Questions

1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#peer-review)?

1. What issue does this PR address?

1. At a high level, how does the fee estimator work?

    (a) What information is recorded for new transactions entering the mempool? (Hint: what does
        [`mapMemPoolTxs`](https://github.com/bitcoin/bitcoin/blob/194710d8ff398838e4e5bb87b56e19ebed1d6c52/src/policy/fees.h#L258) store?)

    (b) What information is recorded for transactions in blocks? (Hint: see [`processBlockTx`](https://github.com/bitcoin/bitcoin/blob/194710d8ff398838e4e5bb87b56e19ebed1d6c52/src/policy/fees.cpp#L596))

    (c) Does the fee estimator track any transactions that are not in the mempool, were removed, or
        were mined but not in the mempool beforehand?

> 1. How does the PR detect which transactions to drop? Do you agree with this approach?

Reviewer: nit, but:
```suggestion
1. In this PR, how do we detect which transactions to drop? Do you agree with this approach?
```
